Load presets and play them with playlist

revmischa opened the pull request:
Support playing presets provided by the user

revmischa committed:
Load presets and play them with playlist

revmischa committed:
CI: build projectM with playlist

revmischa committed:
CI: build projectM with playlist

revmischa committed:
CMAKE_MODULE_PATH

revmischa committed:
projectM4Playlist cmake file

revmischa committed:
projectM4Playlist cmake file

revmischa committed:
projectM4Playlist cmake file

revmischa committed:
projectM4Playlist cmake file

hack-s commented:
is the `.idea` folder being added intentionally ? I would usually not commit the IDE project files..

revmischa commented:
> is the `.idea` folder being added intentionally ? I would usually not commit the IDE project files..

It was intentional, the cmake settings and target apply to anyone using intellij

hack-s commented on src/projectm.c:
```diff
@@ -34,6 +36,11 @@ projectm_handle projectm_init(GstProjectM *plugin)
 
+    // initialize preset playlist
+    playlist = projectm_playlist_create(handle);
+    projectm_playlist_set_shuffle(playlist, true);
```
I think it would be good to expose this as a property in GstProjectM / plugin.h

revmischa replied:
Okay I added the property

hack-s commented on src/projectm.c:
```diff
@@ -84,6 +93,10 @@ projectm_handle projectm_init(GstProjectM *plugin)
+
+        // kick off the first preset
+        if (projectm_playlist_size(playlist) > 0 && ! plugin->preset_locked)
+        projectm_playlist_play_next(playlist, true);
```
This might just be my personal preference...I always use curly brackets for blocks to make the code more readable

revmischa replied:
done

revmischa committed:
Add shuffle presets property

revmischa committed:
Add shuffle presets property

revmischa committed:
Add shuffle presets property

blaquewithaq commented:
I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.

I tested this and it wouldn't work with my current setup so had to go back to master branch.

revmischa commented:
> I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.
> 
> I tested this and it wouldn't work with my current setup so had to go back to master branch.

It is an option now, it's on by default, did you try adding `shuffle-presets=false`?

blaquewithaq commented:
> > I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.
> > I tested this and it wouldn't work with my current setup so had to go back to master branch.
> 
> It is an option now, it's on by default, did you try adding `shuffle-presets=false`?



> > I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.
> > I tested this and it wouldn't work with my current setup so had to go back to master branch.
> 
> It is an option now, it's on by default, did you try adding `shuffle-presets=false`?

Shuffling presets has to do with randomly selecting presets.

There should be an option: enable-playlist, added that disables/enables the ENTIRE feature.

revmischa committed:
Add option to disable playlist entirely

revmischa committed:
retry CI

kblaschke commented:
The playlist headers need an update, we should release v4.1.5 and wait for Homebrew to update the formula. Then the build will pass.

revmischa committed:
retry CI

revmischa committed:
merge

revmischa committed:
formatting

revmischa committed:
gitignore

revmischa committed:
debug

revmischa commented:
> > > I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.
> > > I tested this and it wouldn't work with my current setup so had to go back to master branch.
> > 
> > 
> > It is an option now, it's on by default, did you try adding `shuffle-presets=false`?
> 
> > > I'd like the playlist feature to be optional as well. Because I do not use the playlists feature.
> > > I tested this and it wouldn't work with my current setup so had to go back to master branch.
> > 
> > 
> > It is an option now, it's on by default, did you try adding `shuffle-presets=false`?
> 
> Shuffling presets has to do with randomly selecting presets.
> 
> There should be an option: enable-playlist, added that disables/enables the ENTIRE feature.

Okay I've added `enable-playlist`!

revmischa committed:
readme
